### DISCUSSION TITLE
multiple activations with no way to fix?

### DISCUSSION THREAD

**Discussion by small-cactus** [dscripka/openWakeWord#141] [General]
I use the model to make a prediction. If the prediction meets a certain threshold, it triggers a function that records a command with speech recognition. After that, it resumes listening for a wake word. However, because the system records in 2-second intervals and I have to pause it for speech recognition, when I restart, it doesn't have new audio to listen to. Then I believe It re-examines the last 80 milliseconds of the previous audio, where it might find the wake word I said earlier, mistakenly activating the function again as if I had repeated the wake word, even though I haven't said anything more.

I specifically cannot record an extra 2-3 seconds of data after the wake word is detected, because that would ruin the experience of the app and cause lots of latency. I also tried changing the frame per buffer length, both short and long did nothing noticeable.

I tried to just refresh the audio stream so it no longer has any data about the wake word that was said previously, and it didn't work, so I don't really know what's going on here. I was thinking the only solution is to unload and reload the model, but that's like an extra 0.5-1.5 seconds added to the latency.

Does anyone know of any way to fix this? Any work around? A conceptual explanation helps too, if specific examples are unavailable.

**Comment by dscripka** [comment]
@small-cactus you are correct, the model is still seeing audio containing the wakeword are you resume, so it then detects again. In a recent update some basic [debounce functionality](https://github.com/dscripka/openWakeWord/blob/c40fe924ffa12e9ddf24a3e5fcdeb4fd58ab07eb/openwakeword/model.py#L251) was added to `model.py`, but this still requires that you continuously feed audio into the openWakeWord model.

My advice in this case is to adjust your program such that openWakeWord is still receiving audio while your speech recognition process is also running. That way both programs are aligned, and always have the most current audio data.

Alternatively, you can manually pass in a 1-2 seconds of audio to the openWakeWord models, processing them as quickly as possible to flush out the prediction buffers. As a last resort, you can use the `model.reset` method ([here](https://github.com/dscripka/openWakeWord/blob/c40fe924ffa12e9ddf24a3e5fcdeb4fd58ab07eb/openwakeword/model.py#L226)) to clear the audio buffers, but in rare cases this approach can still fail.
